WordPress/gutenberg
# Media & Text block: Remove the link option when the featured image is used

## What & Why & How?

When the Media & Text block is linked and uses the featured image, the image is not linked on the front of the site.

Originally, this PR was meant to make the link work.
The preferred way to solve the bug, as expressed in the reviews below, would be to update the LinkControl component and replace ImageURLInputUI.

It was then recommended to solve the bug short-term by not showing the link option if the featured image is used, so that
the user can not use the broken control.
The PR was updated to reflect this.

## Updated testing instructions:

1. Activate Gutenberg
2. Create a new post and add a featured image.
3. Add a Media & Text block and select the option "Use featured image".
4. In the block toolbar, confirm that there is no link option.
5. Select another image from the media library.
6. In the block toolbar, confirm that there is a link option, set a link. Save and view the front, confirm that the link is working.
7. Select the featured image again, save and view the front. confirm that there is no link being output. This step is meant to to test that the link that was set in step 6 was properly unset (reset) when the featured image was selected.

<details><summary> Old description</summary>
The pull request updates where in the block markup the `img` tag is inserted:

1. It checks wether the block `$content` contains an `<a>` tag.
2. If there is an `<a>` tag, it inserts the  `<img>` tag by replacing the closing `</a>` tag with  `<img></a>`.
By replacing the closing `</a>` instead of the opening tag, the class names and attributes on the link are preserved.
3. If the block `$content` does not contain a link, the `<img>` tag is inserted directly inside the `<figure>`.


Closes https://github.com/WordPress/gutenberg/issues/60497

## Limitations
The PR does not address the problem that users can not set _individual_ links on the block when it is placed inside a query loop.

## Testing Instructions

1. Activate Gutenberg
2. Create a new post and add a featured image.
3. Add a Media & Text block and select the option "Use featured image".
4. In the block toolbar, select the link option and add a link.
5. Save and view the front.
6. Confirm that the image is linked.
</details>
